### PR TITLE
Add support for lowercase origin header

### PR DESCRIPTION
**Pull request by touzoku**

### CODE REVIEW

File: src/middlewares/cors.js [hunk 1]
```diff
@@ -7,15 +7,16 @@ const defaults = {
...
+    if (originHeader && options.origins.includes(originHeader)) {
+      return originHeader
     } else {
       return options.origins[0]
     }
   } else {
-    if (handler.event.headers.hasOwnProperty('Origin') && options.credentials && options.origin === '*') {
-      return handler.event.headers.Origin
+    if (originHeader && options.credentials && options.origin === '*') {
+      return originHeader
```
Comment (touzoku) [inline src/middlewares/cors.js]: Please note, I haven't changed the original behavior, but this is a security vulnerability. Asterisk origin should never work with credentials! I can see why it seemed convenient, but it the library should gracefully fail instead of hacking the standard CORS behaviour.

**Comment by lmammino** [comment]
Hello @touzoku,
First of all thanks for your amazing contribution!

A few points:
 - I am happy to merge this and I think it adds some good value, but it would be great if you could update the tests to cover for the new edge cases you are addressing now.
- Also, the header management is generally done through a dedicated middleware: [`httpHeaderNormalizer`](https://middy.js.org/docs/middlewares.html#httpheadernormalizer)
- This middleware is one of the significant breaking changes between current 0.x and 1.0.0-alpha. In 1.0.0 all headers are normalized to lowercase, to be consistent with the behaviour of Node.js core `http` module and make middlewares and other code easier to port from other existing codebases.

Let us know what you think and looking forward to getting your contribution merged

**Comment by willfarrell** [comment]
@touzoku Any luck on the requested changes?

**Comment by touzoku** [comment]
> * I am happy to merge this and I think it adds some good value, but it would be great if you could update the tests to cover for the new edge cases you are addressing now.

I have added a test covering this change.

> * Also, the header management is generally done through a dedicated middleware: [`httpHeaderNormalizer`](https://middy.js.org/docs/middlewares.html#httpheadernormalizer)
> * This middleware is one of the significant breaking changes between current 0.x and 1.0.0-alpha. In 1.0.0 all headers are normalized to lowercase, to be consistent with the behaviour of Node.js core `http` module and make middlewares and other code easier to port from other existing codebases.

Well, most of your middleware does account for both lowercase and uppercase: https://github.com/middyjs/middy/blob/0.x/src/middlewares/httpMultipartBodyParser.js#L20

**Review by willfarrell — APPROVED** [review]

**Comment by willfarrell** [comment]
a redundant PR as already been merged in
